core/src/test/java/feign/BaseBuilderTest.java

```diff
@@ -67,4 +67,27 @@ void checkEnrichTouchesAllBuilderFields()
             .responseInterceptor((ic, c) -> c.next(ic)),
         12);
   }
+
+  @Test
+  void checkCloneDontLooseInterceptors() {
+    Feign.Builder originalBuilder = Feign.builder().addCapability(new Capability() {});
+
+    // There is no any interceptors
+    assertThat(originalBuilder.requestInterceptors).isNotNull().isEmpty();
+
+    Feign.Builder enrichedBuilder = originalBuilder.enrich();
+    Feign.Builder enrichedBuilderWithInterceptor = enrichedBuilder.requestInterceptor((req) -> {});
```

Comment:
Well, by design `enrich` is meant to be the last thing you invoke.

After you set all your interceptors, encoders, decodes, you `build`, and build calls `enrich` witch will apply modifications to client, encoder and other components.

Doing this you are messing around with the Capability process.


If you wanna see my point, change you Capability on line 73 to do something with the `requestInterceptor`.

You will see that the capability will be ignored making very hard to anyone to understand why the capability is not working.

Now I don't think we can't have this change

Reply:
Yes, @velo 

I know it. All interceptors added after `enrich` was invoked will be not processed by `Capability`. And this is exactly what my module `oauth2` does: https://github.com/hosuaby/feign/blob/feature/oauth2/oauth2/src/main/java/feign/auth/oauth2/OAuth2Authentication.java#L80

This `OAuth2Authentication` is a `Capability` that adds interceptor, retryer and errorDecoded just before `Feign` is built. So they are ignored by any other capability.

Even if you approved my PR already, please follow the link, and check once again. Do you agree with this design?